### PR TITLE
feat: Appendix Operators and Symbols

### LINKED ISSUE #136: feat: Appendix operators and symbols
Cairo newcomers have a hard time figuring out which symbols they can use. This appendix will list the symbols available in Cairo, and what they correspond to.

For example, `||` and `&&` (Short-circuiting logical OR/AND) are not available, but `|` and `&` (Bitwise OR / AND) are.

The expected output is similar to the Rust book's [operators appendix](https://doc.rust-lang.org/book/appendix-02-operators.html) page.

Estimated time: 2 to 3 hours (this requires you to dig a bit in the compiler code and test multiple symbols)

### PR DESCRIPTION
**Pull request by zarboq**
Close #136

### CODE REVIEW

File: src/appendix-02-operators-and-symbols.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+| `+=` | `var += expr` | Arithmetic addition and assignment | `AddEq` |
+| `,` | `expr, expr` | Argument and element separator | |
+| `-` | `-expr` | Arithmetic negation | `Neg` |
+| `-` | `expr - expr` | Arithmetic subtraction | `Sub` |
+| `-=` | `var -= expr` | Arithmetic subtraction and assignment | `SubEq` |
+| `->` | `fn(...) -> type`, <code>&vert;...&vert; -> type</code> | Function and closure return type | |
+| `.` | `expr.ident` | Member access | |
+| `/` | `expr / expr` | Arithmetic division | `Div` |
+| `/=` | `var /= expr` | Arithmetic division and assignment | `DivEq` |
+| `:` | `pat: type`, `ident: type` | Constraints | |
```
Comment (enitrat) [inline src/appendix-02-operators-and-symbols.md:32]: what is `pat` ?

File: src/appendix-02-operators-and-symbols.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+The following list contains all symbols that are not used as operators; that is, they do not have the same behavior as a function or method call.
+
+Table B-2 shows symbols that appear on their own and are valid in a variety of locations.
+
+<span class="caption">Table B-2: Stand-Alone Syntax</span>
+
+| Symbol | Explanation |
+|--------|-------------|
+| `...u8`, `...i32`, `...usize`, etc. | Numeric literal of specific type |
```
Comment (enitrat) [inline src/appendix-02-operators-and-symbols.md:55]: `i32` ?

File: src/appendix-02-operators-and-symbols.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+| `'...'` | String literal |
```
Comment (enitrat) [inline src/appendix-02-operators-and-symbols.md:56]: let's call it `short string`

File: src/appendix-02-operators-and-symbols.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+| `_` | “Ignored” pattern binding; also used to make integer literals readable |
+
+Table B-3 shows symbols that are used within the context of a module hierarchy path to access an item.
+
+<span class="caption">Table B-3: Path-Related Syntax</span>
+
+| Symbol | Explanation |
+|--------|-------------|
+| `ident::ident` | Namespace path |
+| `::path` | Path relative to the crate root (i.e., an explicitly absolute path) |
```
Comment (enitrat) [inline src/appendix-02-operators-and-symbols.md:66]: not sure about this one. Got an example?

File: src/appendix-02-operators-and-symbols.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+| `enum ident<...> ...` | Define generic enumeration |
+| `impl<...> ...` | Define generic implementation |
+
+Table B-5 shows symbols that appear in the context of calling or defining macros and specifying attributes on an item.
+
+<span class="caption">Table B-5: Macros and Attributes</span>
+
+| Symbol | Explanation |
+|--------|-------------|
+| `#[meta]` | Outer attribute |
```
Comment (enitrat) [inline src/appendix-02-operators-and-symbols.md:89]: what is this one? it's not very clear

File: src/appendix-02-operators-and-symbols.md (hunk 1 above)
Comment (zarboq) [reply src/appendix-02-operators-and-symbols.md:32]: Stands for pattern, it contains identifiers but can also be something like (x, y): (u32, u32)

File: src/appendix-02-operators-and-symbols.md (hunk 5 above)
Comment (zarboq) [reply src/appendix-02-operators-and-symbols.md:89]: Meta is for metadata and it means everything that is giving instructions to the compiler. derive and stuff like that

File: src/appendix-02-operators-and-symbols.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+The following list contains all symbols that are not used as operators; that is, they do not have the same behavior as a function or method call.
+
+Table B-2 shows symbols that appear on their own and are valid in a variety of locations.
+
+<span class="caption">Table B-2: Stand-Alone Syntax</span>
+
+| Symbol | Explanation |
+|--------|-------------|
+| `...u8`, `...usize`, etc. | Numeric literal of specific type |
```
Comment (enitrat) [inline src/appendix-02-operators-and-symbols.md:55]: actually I think it's `_u8`

**Review by enitrat — APPROVED** [review]
fix & go, thanks !

**Pull request merged**